web/client/plugins/widgets/getWidgetFilterRenderers.js

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2019, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+*/
+
+const { connect } = require('react-redux');
+const { createSelector } = require('reselect');
+
+const { getAttributeFields } = require('../../utils/FeatureGridUtils');
+const { getFilterRenderer } = require('../../components/data/featuregrid/filterRenderers');
+const { getWidgetAttributeFilter } = require('../../selectors/widgets');
+
+
+const getWidgetFilterRenderers = createSelector(
```

Comment:
You have all the information as properties for the widget, no need to connect anything.
I should create a withFilters enhancer that parses the describefeauretype and provides the proper filterRenderer at the contained Component. If you want to cache, you can use withPropsOnChange.

Here you're using a selector that is not state selector but a function... it's very weird.

Reply:
ok i'll change it, but it's quite funny how  you call "weird" [your](https://github.com/geosolutions-it/MapStore2/blame/c33eadbc5f900b2b26ca9f18a888ff2dbd3272f9/web/client/plugins/featuregrid/panels/index.jsx#L137) creations :)